spring-integration-core/src/test/java/org/springframework/integration/support/management/graph/IntegrationGraphServerTests.java

```diff
@@ -165,6 +171,41 @@ public PollerMetadata defaultPoller() {
 			return poller;
 		}
 
+		@Bean
+		@Router(inputChannel = "four")
+		public HeaderValueRouter router() {
+			HeaderValueRouter router = new HeaderValueRouter("foo");
+			router.setChannelMapping("bar", "barChannel");
+			router.setChannelMapping("baz", "bazChannel");
+			router.setDefaultOutputChannel(discards());
+			return router;
+		}
+
+		@Bean
+		@Router(inputChannel = "four")
+		public RecipientListRouter rlRouter() {
+			RecipientListRouter router = new RecipientListRouter();
+			router.setRecipients(
+					Arrays.asList(new Recipient[] { new Recipient(barChannel()), new Recipient(bazChannel()) }));
```

Comment:
Ah, yess - I tried 2x `addRecipient(String)` first and had problems with context initialization - those set methods cause early channel resolution - which needs a BF - which hasn't been injected yet.

This class is not friendly for java config - but I didn't notice that setter.